app/models/manageiq/providers/azure/inventory/parser/cloud_manager.rb

```diff
@@ -70,12 +70,15 @@ def instances
 
       rg_ems_ref = collector.get_resource_group_ems_ref(instance)
 
+      # We want to archive VMs with no status
+      next if (status = collector.power_status(instance)).blank?
```

Comment:
do you want to skip all the rest of code or simply want to set the `raw_power_state` to be `nil`?

Reply:
point is to skip ```persister.vms.build``` which will cause the archival of the existing Vm